firefox-ios/Client/Application/AppDelegate+PushNotifications.swift

```diff
@@ -49,37 +49,20 @@ extension AppDelegate {
             }
         }
 
-        // If we see our local device with a pushEndpointExpired flag, clear the APNS token and re-register.
+        // If we see our local device with a pushEndpointExpired flag, try to re-register.
         NotificationCenter.default.addObserver(
             forName: .constellationStateUpdate,
             object: nil,
             queue: nil
         ) { notification in
             if let newState = notification.userInfo?["newState"] as? ConstellationState {
                 if newState.localDevice?.pushEndpointExpired ?? false {
-                    MZKeychainWrapper.sharedClientAppContainerKeychain.removeObject(
-                        forKey: KeychainKey.apnsToken,
-                        withAccessibility: MZKeychainItemAccessibility.afterFirstUnlock
-                    )
                     NotificationCenter.default.post(name: .RegisterForPushNotifications, object: nil)
                     // Our endpoint expired, we should check for missed messages
                     self.profile.pollCommands(forcePoll: true)
                 }
             }
         }
-
-        // Use sync event as a periodic check for the apnsToken.
-        // The notification service extension can clear this token if there is an error,
-        // and the main app can detect this and re-register.
-        NotificationCenter.default.addObserver(forName: .ProfileDidStartSyncing, object: nil, queue: .main) { _ in
-            let kc = MZKeychainWrapper.sharedClientAppContainerKeychain
-            if kc.string(
-                forKey: KeychainKey.apnsToken,
-                withAccessibility: MZKeychainItemAccessibility.afterFirstUnlock
-            ) == nil {
-                NotificationCenter.default.post(name: .RegisterForPushNotifications, object: nil)
-            }
-        }
```

Comment:
this probably led to users having to update their push notification every sync because the apns token will be nil for any user who didn't migrate from the before times.

We don't need to try to re-register for push notifications every sync, there are already recovery mechanisms if a user's push subscriptions change